lib/idempotency/base.rb

```diff
@@ -0,0 +1,85 @@
+# frozen_string_literal: true
+
+require 'base64'
+require_relative 'constants'
+
+module Idempotency
+  module Base
```

Comment:
Could we use composition instead of inheritance to share the code between Rails & Hanami integration? Could the class acts like a rack middleware? I imagine something like this:

```ruby
def use_cache(request_identifiers = [], lock_duration: nil)
  status, headers, body = Idempotency::SomeClass.new.call(request, request_identifiers:, lock_duration:) do
    yield
  end

  # this part is specific for each framework
end
```